### PR TITLE
Create builder projects table

### LINKED ISSUE #22444: Connector builder projects: Add project table
Add a new table `connector_builder_project` to the config postgres database via migration:
* id (PK)
* workspace_id
* name (string)
* manifest_draft (JSON, nullable)
* actor_definition_id (nullable)

The config repository and config writer should be extended with methods for working with builder project objects:
* Read builder project by id
* Read all builder projects by workspace id
* Delete builder project
* Write builder project (create if it doesn't exist, update if it exists)

There should be no foreign key relationships to the actor_definition table to allow splitting out these tables in a future step.
## Acceptance criteria

* Table is created via migration
* Helpers for reading, writing and deleting builder projects are available
* Helpers are unit tested
* Index on workspace_id exists

### PR DESCRIPTION
**Pull request by flash1293**
Fixes https://github.com/airbytehq/airbyte/issues/22444

By introducing the `connector_builder_project` table and adding methods to read/write/delete projects to the `ConfigRepository` class, including tests for these methodes.

Not sure about the version for the platform - is this worth a minor version bump or should it be patch?

## How to test

* Run the tests introduced as part of the PR
* Build platform and start via docker, then log into internal database via `docker exec -ti airbyte-db psql -U docker -d airbyte` and inspect created table.

### CODE REVIEW

File: airbyte-config/config-models/src/main/resources/types/ConnectorBuilderProject.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+"$schema": http://json-schema.org/draft-07/schema#
+"$id": https://github.com/airbytehq/airbyte/blob/master/airbyte-config/models/src/main/resources/types/ConnectorBuilderProject.yaml
+title: ConnectorBuilderProject
+description: low-code cd project editable in the connector builder UI
+type: object
+required:
+  - name
+  - builderProjectId
```
Comment (girarda) [inline airbyte-config/config-models/src/main/resources/types/ConnectorBuilderProject.yaml:9]: i don't know if this is a requirement, but the row's identifier is usually just `id`

File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java [hunk 2]
```diff
@@ -1600,6 +1602,73 @@ public boolean getConnectionHasAlphaOrBetaConnector(final UUID connectionId) thr
...
+
+      if (isExistingConfig) {
+        ctx.update(CONNECTOR_BUILDER_PROJECT)
+            .set(CONNECTOR_BUILDER_PROJECT.ID, builderProject.getBuilderProjectId())
+            .set(CONNECTOR_BUILDER_PROJECT.WORKSPACE_ID, builderProject.getWorkspaceId())
+            .set(CONNECTOR_BUILDER_PROJECT.NAME, builderProject.getName())
+            .set(CONNECTOR_BUILDER_PROJECT.ACTOR_DEFINITION_ID, builderProject.getActorDefinitionId())
+            .set(CONNECTOR_BUILDER_PROJECT.MANIFEST_DRAFT, JSONB.valueOf(Jsons.serialize(builderProject.getManifestDraft())))
+            .set(WORKSPACE.UPDATED_AT, timestamp)
+            .where(getBuilderProjectIdCondition(builderProject.getBuilderProjectId(), builderProject.getWorkspaceId()))
```
Comment (girarda) [inline airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java:1650]: the primary key is the project ID so we shouldn't need the workspace ID in the where clause

**Review by girarda — APPROVED** [review]

File: airbyte-config/config-models/src/main/resources/types/ConnectorBuilderProject.yaml (hunk 1 above)
Comment (flash1293) [reply airbyte-config/config-models/src/main/resources/types/ConnectorBuilderProject.yaml:9]: it is in the table, but it doesn't have to be in the config object passed around in the application layer (e.g. https://github.com/airbytehq/airbyte/blob/b9e3c8a7f26b5fb108cc63a6f2a9878ad062547d/airbyte-config/config-models/src/main/resources/types/StandardSourceDefinition.yaml#L16 even though the column in the database is `id`)

File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java (hunk 2 above)
Comment (flash1293) [reply airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java]: good point, changed

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Review by xiaohansong — APPROVED** [review]

**Comment by flash1293** [comment]
create-platform-pr

**Pull request closed without merging**